File: tests/characterization/fixtures/tier2/pricing.rb
require 'json'

TAX_RATE = 0.2

def apply_tax(amount, rate = TAX_RATE)
  # The tax is applied on the whole subtotal.
  amount * (1 + rate)
end

class Invoice
  def initialize(subtotal)
    @subtotal = subtotal
  end

  def price
    apply_tax(@subtotal, TAX_RATE)
  end
end

File: tests/characterization/fixtures/tier2/pricing.lua
local money = require("app.money")

local TAX_RATE = 0.2

local function apply_tax(amount, rate)
  -- The tax is applied on the whole subtotal.
  return amount * (1 + (rate or TAX_RATE))
end

local Invoice = {}

function Invoice.price(subtotal)
  return apply_tax(subtotal, TAX_RATE)
end

return Invoice
